### PR TITLE
nixos/tests/gnome3: port to python

### PR DESCRIPTION
**Pull request by x123**
###### Motivation for this change
Port `gnome3` test to python for https://github.com/NixOS/nixpkgs/issues/72828

###### Things done
Ported `gnome3` tests to python.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) `nixos/tests/gnome3.nix`
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by worldofpeace** [comment]
Hi @x123, you may haven't seen in #72828, but I've actually taken on porting all the DE tests. And this one was probably done first, I just haven't PR'd it.

**Comment by x123** [comment]
@worldofpeace Oh, alright....but how was I supposed to know if they weren't PR'ed or documented anywhere in the issue? I guess I'll stop trying to port any tests if it's not worth doing due to undocumented work duplication. Thanks for the heads up.

**Comment by worldofpeace** [comment]
We followed up on IRC.

Screenshot from 2019-12-03 17 27 44

File: nixos/tests/gnome3.nix [hunk 1]
```diff
@@ -38,27 +38,36 @@ import ./make-test.nix ({ pkgs, ...} : {
...
+
+      machine.sleep(5)
 
       # wait for alice to be logged in
-      $machine->waitForUnit("default.target","alice");
+      machine.wait_for_unit("default.target", "alice")
 
       # Check that logging in has given the user ownership of devices.
-      $machine->succeed("getfacl -p /dev/snd/timer | grep -q alice");
+      machine.succeed("getfacl -p /dev/snd/timer | grep -q alice")
```
Comment (tfc) [inline nixos/tests/gnome3.nix]: how about `assert "alice" in machine.succeed("...")`

Comment (x123) [reply nixos/tests/gnome3.nix]: @tfc done.

**Pull request closed without merging**